cmflib/minio_artifacts.py

```diff
@@ -0,0 +1,58 @@
+import os
+from minio import Minio
+from minio.error import S3Error
+
+
+class minio_artifacts:
+    def get_dvc_config(self, current_directory: str):
+        dvc_path = current_directory + "/.dvc/config"
+        if os.path.exists(dvc_path):
+            file = open(dvc_path, "r")
+            while True:
+                line = file.readline()
+                if not line:
+                    break
+                if line.find("endpointurl") != -1:
+                    temp = line.split()[2]
+                    endpoint = ''.join(temp.split("http://", 1))
+                elif line.find("access_key_id") != -1:
+                    access_key = line.split()[2]
+                elif line.find("secret_access_key") != -1:
+                    secret_key = line.split()[2]
+            return endpoint, access_key, secret_key
+        else:
+            pass
+
+    def download_artifacts(self, current_directory: str, bucket_name: str, object_name: str, file_path: str):
+        endpoint = ""
+        access_key = ""
+        secret_key = ""
+        try:
+            endpoint, access_key, secret_key = self.get_dvc_config(current_directory)
+            client = Minio(
+                endpoint,
+                access_key=access_key,
+                secret_key=secret_key,
+                secure=False
+            )
+            found = client.bucket_exists("dvc-art")
```

Comment:
bucket name should not be assumed , the user can have different bucket name

Reply:
Made the required change